core/player/src/controllers/error/controller.ts

```diff
@@ -0,0 +1,229 @@
+   * - Called in order for each tapped plugin
+   * - Return true to bail and prevent error state navigation
+   * - Return undefined/false to continue to next handler
+   * - Once true is returned, no further plugins are called
+   */
+  onError: SyncBailHook<[PlayerError], boolean | undefined>;
+}
+
+export interface ErrorControllerOptions {
+  /** Optional logger for error operations */
+  logger?: Logger;
+  /** Data model for setting errorState */
+  model?: DataController;
+}
+
+/**
+ * Get the middleware for protecting errorState from external writes
+ * Should be added to DataController's middleware array
+ */
+export function getErrorStateMiddleware(options?: {
+  logger?: Logger;
+}): ErrorStateMiddleware {
+  return new ErrorStateMiddleware(options);
+}
+
+/** Default error format for data.errorState  */
+interface FormattedError {
+  message: string;
+  name: string;
+  errorType?: string;
+  severity?: ErrorSeverity;
+  assetId?: string;
+  assetType?: string;
+  bindingPath?: string;
+  context?: Record<string, unknown>;
+}
+
+/** The orchestrator for player error handling */
+export class ErrorController {
+  public hooks: ErrorControllerHooks = {
+    onError: new SyncBailHook<[PlayerError], boolean | undefined>(),
+  };
+
+  private options?: ErrorControllerOptions;
+  private readonly middleware: ErrorStateMiddleware;
+  /**
+   * Complete history of all captured errors in chronological order
+   * Newest errors are APPENDED to the end of the array
+   */
+  private errorHistory: PlayerError[] = [];
+  private currentError?: PlayerError;
+
+  constructor(options: ErrorControllerOptions = {}) {
+    this.options = options;
+
+    this.middleware = getErrorStateMiddleware({ logger: options.logger });
+  }
+
+  /**
+   * Get the middleware for protecting errorState
+   * This should be added to DataController's middleware array
+   */
+  public getDataMiddleware(): ErrorStateMiddleware {
+    return this.middleware;
+  }
+
+  /**
+   * Set options after initialization (e.g., to inject DataController and logger)
+   */
+  public setOptions(options: ErrorControllerOptions): void {
+    this.options = options;
+  }
+
+  /**
+   * Capture error with metadata, add to history, fire hooks, update data model
+   */
+  public captureError(error: Error, metadata: ErrorMetadata = {}): PlayerError {
+    const playerError: PlayerError = {
+      error,
+      metadata: {
+        ...metadata,
+        timestamp: metadata.timestamp ?? Date.now(),
+      },
+    };
+
+    // Add to history
+    this.errorHistory.push(playerError);
+
+    // Set as current error
+    this.currentError = playerError;
+
+    this.options?.logger?.debug(
+      `[ErrorController] Captured error: ${error.message}`,
+      playerError.metadata,
+    );
+
+    // Notify listeners and check if navigation should be skipped
+    // Plugins can observe the error and optionally return true to bail
+    const shouldSkip = this.hooks.onError.call(playerError) ?? false;
```

Comment:
if we're using `false` here should the type on the hook be boolean without undefined?